src/Psalm/Internal/Fork/Pool.php

```diff
@@ -354,6 +354,10 @@ private function readResultsFromChildren(): array
                                 ($this->task_done_closure)($message->data);
                             }
                         } elseif ($message instanceof ForkProcessErrorMessage) {
+                            // Kill all children
+                            foreach ($this->child_pid_list as $child_pid) {
+                                posix_kill($child_pid, \SIGKILL);
```

Comment:
Wouldn't `SIGTERM` be more appropriate?

Also, you need to reap child exit statuses with [`pcntl_waitpid()`](php.net/pcntl_waitpid) lest they become zombies.

Reply:
the code to do that is already there in `Pool::wait()`, but it's not executed when stack is unwinding due to exception. It may make sense to wrap it in `try ... finally` and reap children there.

Reply:
As long as the parent process dies the zombies should be taken care of by init, is there a reason to handle that ourselves instead?

SIGTERM is probably more appropriate, I just defaulted to SIGKILL because I wanted to be certain the children were killed.

Reply:
That assumes there is init, and it's often absent in container environments. If memory serves PHPStorm plugin runs Psalm in a container - but I don't really know how exactly they do it.

Reply:
It looks like that's an issue that should be solved by the `--init`option to `docker run`, but I suppose we might as well reap them here in case someone decides to run it in a container without an init process. I'll update it tomorrow.